kategory/src/main/kotlin/kategory/instances/WriterTInstances.kt

```diff
@@ -1,31 +1,104 @@
 package kategory
 
-interface WriterTInstances<F, W> :
-        Functor<WriterTKindPartial<F, W>>,
-        Applicative<WriterTKindPartial<F, W>>,
-        Monad<WriterTKindPartial<F, W>>,
-        MonadWriter<WriterTKindPartial<F, W>, W> {
+interface WriterTInstances : WriterTInstances0 {
 
-    fun MM(): Monad<F>
+    fun <F, W> monadForWriterT(MF: Monad<F>, MW: Monoid<W>): Monad<WriterTKindPartial<F, W>> =
```

Comment:
Not exactly. Those interfaces are named, instead of directly using the anonymous classes created inside the `WriterT` companion. Thanks to this I am able to apply inheritance relationship between all the interfaces to compose the instances behavior, inherited by the companion.

The companion methods are just shortcuts to this, so they can pass injected global monad instances to them and keep the inheritance relationship between all of them.

Reply:
See https://github.com/kategory/kategory/pull/242/files/2d736d0782b2d36ed4b730ba668030fff5d2ec19#r136718720 , I still see no point for an extra layer of indirectionin Kotlin because we have no implicit resolution that justifies factories in each instance declaration and the companion. I may be missing something about your point on inheritance relationship between the interfaces. If you want to elaborate on that piece I'll appreciate it to understand what the benefit of these methods are.

Reply:
In Scala those methods exist as implicits to enable the Lower Priority Implicits pattern but we don;t need that in Kotlin https://stackoverflow.com/questions/33544212/explain-the-lowpriorityimplicits-pattern-used-in-scala-type-level-programming
Scala needs those to ensure user's provided implicits don't cause ambiguity and collisions when the compiler resolves implicits since those in lower traits will have less precedence.